### PR TITLE
NRF5: sleep() improvements

### PR DESCRIPTION
**Pull request by d00616**
Sleep(0) results in no sleep without this fix.

### CODE REVIEW

**Comment by d00616** [comment]
Jenkins, build examples please

**Comment by d00616** [comment]
Jenkins, build examples please

**Comment by d00616** [comment]
Jenkins, build examples please

File: hal/architecture/MyHwNRF5.cpp [hunk 1]
```diff
@@ -204,16 +207,25 @@ void hwSleepPrepare(unsigned long ms)
...
+			// Set compare register to 1/30.517 µs to garantee event triggering
+			// A minimum of 2 ticks must be guaranteed
+			// (1000/32768)<<12 == 125
+			MY_HW_RTC->CC[0] = min((ms<<12 / 125), 2);
+		} else {
+			// 8 Hz -> max 582.542 hours sleep.
+			MY_HW_RTC->PRESCALER = 4095;
+			// Set compare register to 1/125ms
+			// A minimum of 2 ticks must be guaranteed
+			MY_HW_RTC->CC[0] = min((ms / 125), 2);
```
Comment (mfalkvidd) [inline hal/architecture/MyHwNRF5.cpp]: The comment says "A minimum of 2 ticks must be guaranteed" but min(a,b) will return the smaller value. Should this line be
MY_HW_RTC->CC[0] = max((ms / 125), 2);
instead?

Comment (d00616) [reply hal/architecture/MyHwNRF5.cpp]: Thank you. I fix this.

**Comment by d00616** [comment]
Jenkins, build examples please

**Comment by d00616** [comment]
Jenkins, build examples please

**Pull request merged**